README.md

```diff
@@ -90,14 +90,45 @@ vault_attribute :credit_card,
 - **Note** This value **cannot** be the same name as the vault attribute!
 
 #### Specifying a custom key
-By default, the name of the key in Vault is `#{app}_#{table}_#{column}`. This is customizable by setting the `:key` option when declaring the attribute:
+
+By default, the name of the key in Vault is `#{app}_#{table}_#{column}`. This is customizable by setting the `:key` option when declaring the attribute.
+
+- **Note** Changing this value for an existing application will make existing values no longer decryptable!
+
+##### String
+
+With a string, all records will use the same transit key for this attribute:
 
 ```ruby
 vault_attribute :credit_card,
   key: "pci-data"
 ```
 
-- **Note** Changing this value for an existing application will make existing values no longer decryptable!
+##### Symbol
+
+When using a symbol, a method will be called on the record to compute the key name:
+
+```ruby
+belongs_to :user
+
+vault_attribute :credit_card,
+  key: :vault_key
```

Comment:
Accepting a symbol is potentially backwards-incompatible, as existing behavior for passing a symbol would be to coerce it to a string.

We could either:

1. Bump the version such that we note this is a breaking change.
1. Check if the object responds to the given method. If not, coerce it to a string and print a warning to stdout, and remove this behavior in a future version.

Reply:
For me, 1) would be a better option. It's true that supporting both as per option 2 would be nicer for people expecting symbols to coerce, but it would be odd for proper uses where symbols are intended to invoke methods, but instead get coerced to string. It seems like in that case it would be possible that a bunch of data would end up getting encrypted using a static key instead of the intended dynamic one.

I think pinning the gem version to retain old behavior is reasonable, and bumping the version up for the new functionality is probably the way to go. Also, just IMO, passing a symbol and expecting/assuming it gets converted to a string is pretty weird, and AFAIK this was just coincidence that it worked, and not an intentional design decision.

Reply:
Here's an example diff of option 2 for posterity: https://gist.github.com/justincampbell/5f2ca1ad78ff3d6ea8eb33bbb7989297

Reply:
Especially given that this gem is still marked < 1.0, I'm also in favor of option 1, and agree with Ryan sentiment about accidental encryption. 

You can also guide users, too - perhaps you should tag on to the expected `NoMethodError` here about this behavior. (to be removed in a few versions).

👍🏻